### PR TITLE
Fixes #2782

### LINKED ISSUE #2782: addHs() fails on atoms with "bad" valences
Here's what currently happens:
```
In [10]: m = Chem.MolFromSmiles('C=C1=CC=CC=C1',sanitize=False)                                                                                                                          

In [11]: m.UpdatePropertyCache(strict=False)                                                                                                                                             

In [12]: Chem.AddHs(m)                                                                                                                                                                   
[16:04:22] Explicit valence for atom # 1 C, 5, is greater than permitted
---------------------------------------------------------------------------
AtomValenceException                      Traceback (most recent call last)
<ipython-input-12-c1c20d587653> in <module>
----> 1 Chem.AddHs(m)

AtomValenceException: Explicit valence for atom # 1 C, 5, is greater than permitted

```
This is because `addHs()` is calling `atom->updatePropertyCache()` with default arguments (which is to be strict).
There's no need for this and it just makes things difficult for people who want to work with molecules where some atoms are in odd valence states.

### PR DESCRIPTION
**Pull request by greglandrum**
Small change to allow AddHs() to be called on molecules with "unhappy" atoms.

### CODE REVIEW

File: Code/GraphMol/catch_tests.cpp [hunk 1]
```diff
@@ -779,3 +779,18 @@ TEST_CASE("setDoubleBondNeighborDirections()", "[stereochemistry,bug]") {
...
+    ps.sanitize = false;
+    std::unique_ptr<RWMol> m(
+        static_cast<RWMol *>(SmilesToMol("C=C1=CC=CC=C1", ps)));
+    REQUIRE(m);
+    bool strict = false;
+    m->updatePropertyCache(strict);
+    CHECK(m->getNumAtoms() == 7);
+    MolOps::addHs(*m);
+    CHECK(m->getNumAtoms() == 14);
+  }
```
Comment (d-b-w) [inline Code/GraphMol/catch_tests.cpp]: It might be reasonable to add a check that sanitization still fails here.

**Comment by d-b-w** [comment]
Looks good to me!

**Comment by bp-kelley** [comment]
@greglandrum I’ll merge when the conflicts are sorted

**Comment by greglandrum** [comment]
> @greglandrum I’ll merge when the conflicts are sorted

grrr.... I missed that. Fixed. Let's see how the CI builds go

**Comment by greglandrum** [comment]
> @greglandrum I’ll merge when the conflicts are sorted

Also: nice use of the passive voice here. ;-)

**Pull request merged**